arwen/contexts/runtime.go

```diff
@@ -48,22 +55,31 @@ type runtimeContext struct {
 
 	validator       *wasmValidator
 	instanceBuilder arwen.InstanceBuilder
-	errors          arwen.WrappableError
+	hasher          arwen.HashComputer
+
+	errors arwen.WrappableError
 }
 
 // NewRuntimeContext creates a new runtimeContext
 func NewRuntimeContext(
 	host arwen.VMHost,
 	vmType []byte,
 	builtInFuncContainer vmcommon.BuiltInFunctionContainer,
+	hasher arwen.HashComputer,
 ) (*runtimeContext, error) {
+
+	if check.IfNil(hasher) {
+		return nil, arwen.ErrNilHasher
```

Comment:
a unit test for this would have been great

Reply:
Added.